yt/yt/server/controller_agent/controllers/task.cpp

```diff
@@ -634,18 +638,27 @@ TTask::GetOutputCookieInfoForFirstJob(const TAllocation& allocation, const TNewJ
 
     TOutputCookieInfo result;
 
-    if (TaskHost_->IsTreeProbing(allocation.TreeId)) {
+    if (MultiJobManager_.GetPendingJobCount() != 0) {
+        auto [cookie, i] = MultiJobManager_.PeekJobCandidate();
+        result.CompetitionType = EJobCompetitionType::Multi;
+        result.OutputCookie = cookie;
+        result.OutputCookieGroupIndex = i;
+    } else if (TaskHost_->IsTreeProbing(allocation.TreeId)) {
         result.CompetitionType = EJobCompetitionType::Probing;
         result.OutputCookie = ProbingJobManager_.PeekJobCandidate();
+        result.OutputCookieGroupIndex = 0;
     } else if (ExperimentJobManager_.IsTreatmentReady()) {
         result.CompetitionType = EJobCompetitionType::Experiment;
         result.OutputCookie = ExperimentJobManager_.PeekJobCandidate();
+        result.OutputCookieGroupIndex = 0;
     } else if (speculative) {
         result.CompetitionType = EJobCompetitionType::Speculative;
+        result.OutputCookieGroupIndex = 0;
         result.OutputCookie = SpeculativeJobManager_.PeekJobCandidate();
     } else {
         result.CompetitionType = std::nullopt;
         result.OutputCookie = ExtractCookieForAllocation(allocation, newJobConstraints);
+        result.OutputCookieGroupIndex = 0;
```

Comment:
I suggest to omit this explicit assignments to zero, since 0 is a default value in the struct